workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -29,48 +42,116 @@ protected override void OnCreateManager()
+                ));
+
+                SendCreatePlayerRequest();
+            }
+            else
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent(
+                    $"Unable to create player after {playerCreationAttempts} attempts."
+                ));
+            }
         }
 
         protected override void OnUpdate()
         {
-            if (PlayerLifecycleConfig.AutoRequestPlayerCreation && !initializationGroup.IsEmptyIgnoreFilter)
+            if (playerCreatorEntityIds.Count > 0)
             {
-                RequestPlayerCreation();
-            }
+                if (playerRequestQueued)
+                {
+                    SendCreatePlayerRequest();
+                }
 
-            // Currently this has a race condition where you can receive two entities
-            // The fix for this is more sophisticated server side handling of requests
-            var responses = commandSystem.GetResponses<PlayerCreator.CreatePlayer.ReceivedResponse>();
+                // Currently this has a race condition where you can receive two entities
+                // The fix for this is more sophisticated server side handling of requests
+                var responses = commandSystem.GetResponses<PlayerCreator.CreatePlayer.ReceivedResponse>();
 
-            for (var i = 0; i < responses.Count; i++)
-            {
-                ref readonly var response = ref responses[i];
-                if (response.StatusCode == StatusCode.AuthorityLost)
+                for (var i = 0; i < responses.Count; i++)
                 {
-                    RetryCreatePlayerRequest();
+                    ref readonly var response = ref responses[i];
+                    if (response.StatusCode == StatusCode.AuthorityLost)
+                    {
+                        RetryCreatePlayerRequest();
+                    }
+                    else if (response.StatusCode != StatusCode.Success)
+                    {
+                        logDispatcher.HandleLog(LogType.Error, new LogEvent(
+                            $"Create player request failed: {response.Message}"
+                        ));
+                    }
                 }
-                else if (response.StatusCode != StatusCode.Success)
+            }
+            else
+            {
+                var entityQueryResponses = commandSystem.GetResponses<WorldCommands.EntityQuery.ReceivedResponse>();
+                for (var i = 0; i < entityQueryResponses.Count; i++)
                 {
-                    logDispatcher.HandleLog(LogType.Error, new LogEvent(
-                        $"Create player request failed: {response.Message}"
-                    ));
+                    ref readonly var response = ref entityQueryResponses[i];
+                    if (response.RequestId == playerCreatorQueryId)
+                    {
+                        if (response.StatusCode == StatusCode.Success && response.Result != null)
```

Comment:
I think one of the contracts from the Worker SDK is that if the response is a success, the Result cannot be null